### ISSUE TITLE
List of supported biomedical objects/instances during training?

### ISSUE THREAD

**Issue by XYPB** [microsoft/BiomedParse#82]
Hi there,

Thank you so much for open-sourcing this amazing work! It works pretty well and the documentation is also nice. 

However, I note some small issues while using the pre-trained v1 model. It seems that the model is not performing very well when prompting with "unseen" instance, or "unseen" image pose. Just like the examples below:

<img width="1000" height="500" alt="Image" src="https://github.com/user-attachments/assets/321b19f3-062e-424c-9d5b-50f0290d00b5" />

<img width="1000" height="500" alt="Image" src="https://github.com/user-attachments/assets/6b36ba44-e706-4b4c-9d00-ba19b36ab449" />

The first example includes an unseen instance, and the second one is a relatively rare image pose sliced from a 3D volume. In both case the model failed to give a reasonable segmentation results. For the first case, the model seems tend to segment something that it already know ("lung" here) rather than the instructed object.

This was kind of annoying since I am not sure in which cases the model is mostly reliable. I notice that there is indeed a list of `BIOMED_OBJECTS` in the `utilities/constants.py`, which has 52 instance names there. 

However, according to the paper, the model was trained on:
> The semantic labels encompass 82 major biomedical object types across 9 imaging modalities.

So I am wondering where can I find the full list of the biomedical object types for the pre-trained model? And have the model been evaluated systematically on instances that it does not know? Like, OOD objects. Or it is mostly recommended to fine-tune it on my data or simply focus on in-domain data?

Thank you in advance! Great work!

Best

---

Edit: The segmentation above was generated using `example_prediction.py` and it works well on the example data, so I assume my results should be correct.

**Comment by theodore-zhao** [comment]
Thanks for your interest in our work! You are correct that the model works for the image and tasks that were used for training. It's recommended to finetune the model on the data if OOD. Here is a list of anatomies for reference https://github.com/microsoft/BiomedParse/blob/main/inference_utils/target_dist.json The complete list of anatomy evaluation can be found here: https://github.com/microsoft/BiomedParse/blob/main/figures/results/all_eval/all_metrics_median.csv

I noticed you are working on 3D CT image. For that purpose we recently released the v2 branch, which includes  a lot more anatomies and views. It would be good to check it out!

**Comment by XYPB** [comment]
Thank you for your promptly reply! That solves my questions.

Have a nice weekend and thank you again for the amazing work~
